app/src/main/java/org/akvo/flow/ui/view/RepeatableGroupIterations.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (C) 2019 Stichting Akvo (Akvo Foundation)
+ *
+ * This file is part of Akvo Flow.
+ *
+ * Akvo Flow is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Akvo Flow is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Akvo Flow.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.akvo.flow.ui.view;
+
+import android.text.TextUtils;
+
+import org.akvo.flow.domain.QuestionResponse;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+
+import androidx.annotation.NonNull;
+
+class RepeatableGroupIterations implements Iterable<Integer> {
```

Comment:
I was not very clear on how this class is going to work but thats better addressed in a call

Reply:
to be honest I have no idea either, I have extracted this, it was an inner class. Will add some tests to it.